### PR TITLE
Move from libfuse2 to libfuse3

### PR DESCRIPTION
**Pull request by ikbenlike**

### CODE REVIEW

File: echfs-fuse.c [hunk 1]
```diff
@@ -600,47 +588,19 @@ static int echfs_opendir(const char *dir_path,
...
-    stat->st_mode = 0;
-    switch (path_result->target.type) {
-        case DIRECTORY_TYPE:
-            stat->st_mode |= S_IFDIR;
-            break;
-        case FILE_TYPE:
-            stat->st_mode |= S_IFREG;
-            break;
+    struct path_result_t *path_result;
+    if (file_info) {
```
Comment (Geertiebear) [inline echfs-fuse.c:595]: Why was all of this removed? Right now it seems this doesn't do anything.

**Review by Geertiebear — COMMENTED** [review]
One comment, for the rest LGTM.

File: echfs-fuse.c (hunk 1 above)
Comment (ikbenlike) [reply echfs-fuse.c:595]: For libfuse2, there were separate functions for the same functionality: one that could work on the file handle, and one that works on the node. In libfuse3 the same function gets called in either case, but this isn't very clear from the diff

Comment (Geertiebear) [reply echfs-fuse.c:595]: Ahh okay, then that sounds good, thought we accidentally lost functionality here.

**Pull request merged**